package.json

```diff
@@ -18,54 +19,22 @@
   "types": "lib/peg.d.ts",
   "bin": "bin/peggy",
   "scripts": {
-    "pretest": "npm run parser",
+    "build": "pnpm run -r --if-present --filter ./pkg build",
+    "pretest": "npm run build",
     "test": "npm run test:bare",
-    "test:bare": "mocha --recursive test --exclude test/browser.stub.js --reporter min",
-    "pretest:node": "npm run parser",
+    "test:bare": "pnpm run -r --if-present --filter ./pkg test",
```

Comment:
as above.  this means i also can't test on work machines 😥

there is no actual value being delivered here.  this can be done without `pnpm`

there is no apparent need to subdivide the base repo.  

this will lead to constant conway churn

Reply:
There is a need to divide the base repo.  The CLI needs to take a node-only dependency to have any hope of being modern, and there's no reason for the web version to have a CLI or take a dependency.

Reply:
> There is a need to divide the base repo.

I do not agree.

&nbsp;

> The CLI needs to take a node-only dependency to have any hope of being modern

So what?  `package.json` is a node-only context.  dependencies from `package.json` don't affect non-cli things.

Just build the CLI in rollup.  It doesn't need access to dependencies after that.  This is a problem with webpack.

.

> there's no reason for the web version to have a CLI or take a dependency.

My core point is that there shouldn't be a "the web version"

The difference between the two is smaller than the packages being used to separate them

100% of the reason given for needing to split them is a result of having already split them

If you just don't split them in the first place, the reason goes away

This would be much simpler in the original single-repo non-monorepo structure

Reply:
This didn't exist at all before futagoza

It worked a lot better back then

This was added because futagoza tried to move the hosting to some knockoff of heroku that made deploys using the git repo as structure, then later off of it when they went out of business

Reply:
> Just build the CLI in rollup. It doesn't need access to dependencies after that.

Now *that* is a good idea.

Reply:
I'll do the labor if you want, have it running a bunch of other places too already